astropy/coordinates/representation/base.py

```diff
@@ -643,6 +656,8 @@ class BaseRepresentation(BaseRepresentationOrDifferential):
     _differentials = {}
 
     def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)  # sets `cls.name`
+
         # Register representation name (except for bases on which other
         # representations are built, but which cannot themselves be used).
         if cls.__name__.startswith("Base"):
```

Comment:
Could avoid giving these a name by putting this before the `super()` (and maybe the check on `attr_classes` too).

Reply:
The "name" (via get_name) was in the promised API before this PR. It would be strange to remove it here.